saleor/graphql/core/mutations.py

```diff
@@ -383,38 +388,60 @@ def user_is_allowed(cls, user, ids):
         """
         return True
 
+    @classmethod
+    def mutate(cls, root, info, **data):
+        if not cls.user_is_allowed(info.context.user, data):
+            raise PermissionDenied()
+
+        count, errors = cls.perform_mutation(root, info, **data)
```

Comment:
I (maybe overreacting) see some danger here. In base case this method return one argument (`None`) and it's not overwritten in this class to return more, so if someone forgets implementing this, may get confused by `TypeError: cannot unpack non-iterable NoneType object`